src/addressResolvers/utils.ts

```diff
@@ -9,12 +9,22 @@ const broviderUrl = process.env.BROVIDER_URL || 'https://rpc.brovider.xyz';
 
 export class FetchError extends Error {}
 
+export function isEvmAddress(address: Address): boolean {
+  return /^0x[a-fA-F0-9]{40}$/.test(address);
+}
+
+export function isNonEvmAddress(address: Address): boolean {
+  return /^0x[a-fA-F0-9]{64}$/.test(address);
+}
+
 export function provider(network: string) {
   return snapshot.utils.getProvider(network, { broviderUrl });
 }
 
-export function withoutEmptyValues(obj: Record<string, any>) {
-  return Object.fromEntries(Object.entries(obj).filter(([, value]) => value));
+export function withoutEmptyValues(obj: any[] | Record<string, any>) {
+  const data = Array.isArray(obj) ? obj : Object.entries(obj);
+
+  return Object.fromEntries(data.filter(([, value]) => value));
 }
```

Comment:
We now accept a array just for starknet? better to convert and pass it from starknet resolver itself than accepting both array and object